target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -160,6 +162,15 @@ ipq40xx_setup_macs()
 		wan_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
 		lan_mac=$(macaddr_add "$wan_mac" 1)
 		;;
+	mikrotik,hap-ac2)
+		wan_mac=$(cat /sys/firmware/mikrotik/hard_config/mac_base)
+		lan_mac=$(macaddr_add $wan_mac 1)
+		label_mac="$wan_mac"
+		;;
+	mikrotik,sxtsq-5-ac)
+		lan_mac=$(cat /sys/firmware/mikrotik/hard_config/mac_base)
+		label_mac="$lan_mac"
+		;;
```

Comment:
If this were part of a subtarget-only file (as I suggested on the hapac2 PR), it would be possible to define mac_base for all devices and use it for label_Mac for all devices as well. I suspect keeping this in the main target will result in lots of duplicated segments. My 2c.
@robimarko

Reply:
I partly agree, but having MikroTik only base-files would incur a lot of duplicates currently.
There are simply not enough MikroTik devices in it to justify duplicating everything.

Also, we can simply define the MAC base once.

Reply:
> I partly agree, but having MikroTik only base-files would incur a lot of duplicates currently.
> There are simply not enough MikroTik devices in it to justify duplicating everything.

There are simply not enough MikroTik devices _now_. It might be useful to anticipate a bit :)